.github/workflows/CI.yml

```diff
@@ -14,41 +14,79 @@
 # limitations under the License.
 
 ---
-name: Code Formatting
+name: Daffodil CI
 on:
   push:
     branches-ignore: [ 'dependabot/**' ]
   pull_request:
     types: [opened, synchronize, reopened]
 
 jobs:
-  ts-format:
-    name: TypeScript Formatting
+  single-commit:
+    name: Single Commit Pull Request
+    if: github.event_name == 'pull_request'
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2.3.5
-      - name: Setup Node
-        uses: actions/setup-node@v1.4.6
+      - name: Check Single Commit
+        uses: actions/github-script@v5.0.0
         with:
-          node-version: '16'
-      - run: yarn install
-      - run: yarn lint
+          script: |
+            const commits = await github.rest.pulls.listCommits({
+              ...context.repo,
+              pull_number: context.issue.number,
+            });
+            core.info("Number of commits in this pull request: " + commits.data.length);
+            if (commits.data.length > 1) {
+              core.setFailed("If approved with two +1's, squash this pull request into one commit");
+            }
 
-  scala-format:
-    name: Scala Formatting
+  format-rat-check:
```

Comment:
Having the TypeScript and Scala formatting in one job I think is fine but the ratCheck should its own separate stage.

Reply:
Agreed.